src/compass/utils/h5_helpers.py

```diff
@@ -193,256 +202,269 @@ def init_geocoded_dataset(geocoded_group, dataset_name, geo_grid, dtype,
+
+    # frequencyA datasets
+    freq_meta_items = [
+        # 'frequencyA/azimuthBandwidth':
+        meta('range_bandwidth', burst.range_bandwidth,
+             'Processed range bandwidth in Hz'),
+        meta('center_frequency', burst.radar_center_frequency,
+             'Center frequency of the processed image in Hz'),
+        meta('slant_range_spacing', burst.range_pixel_spacing,
+             'Slant range spacing of grid. '
+             'Same as difference between consecutive samples in slantRange array'),
+        meta('zeroDoppler_time_spacing', burst.azimuth_time_interval,
+             'Time interval in the along track direction for raster layers. This is same '
+             'as the spacing between consecutive entries in the zeroDopplerTime array')
+    ]
+    freq_group = dst_h5.require_group('/CSLC/grids/frequencyA')
+    for meta_item in freq_meta_items:
+        add_dataset_and_attrs(freq_group, meta_item)
+
+
+def burst_corrections_to_hdf5(dst_h5, burst, cfg):
+    '''
+    Write azimuth, slant range, and EAP (if needed) correction LUT2ds to HDF5
+
+    Parameter:
+    ---------
+    dst_h5: h5py File
+        HDF5 file meta data will be written to
+    burst: Sentinel1BurstSlc
+        Burst containing corrections
+    cfg: types.SimpleNamespace
+        SimpleNamespace containing run configuration
+    '''
+    # Get range and azimuth LUTs
+    rg_lut, az_lut = compute_geocoding_correction_luts(burst,
+                                                       rg_step=cfg.lut_params.range_spacing,
+                                                       az_step=cfg.lut_params.azimuth_spacing)
+
+    # az/rg_lut share the same grid
+    slant_range = np.linspace(az_lut.x_start,
+                              az_lut.x_start + az_lut.width * az_lut.x_spacing,
+                              az_lut.width, dtype=np.float64)
+    azimuth = np.linspace(az_lut.y_start,
+                          az_lut.y_start + az_lut.width * az_lut.x_spacing,
+                          az_lut.width, dtype=np.float64)
+
+    # correction LUTs axis
+    correction_axis_items = [
+        meta('slant_range', slant_range, 'slant range of LUT data',
+             {'units': 'meters'}),
+        meta('azimuth_time', azimuth, 'azimuth time of LUT data',
+             {'units': 'seconds'})
+    ]
+    correction_axis_group = dst_h5.require_group('/CSLC/grids/corrections')
+    for meta_item in correction_axis_items:
+        add_dataset_and_attrs(correction_axis_group, meta_item)
+
+    # doppler correction LUTs
+    desc = ' correction as a function of slant range and azimuth time'
+    doppler_correction_items = [
+        meta('azimuth_LUT', az_lut.data, f'azimuth {desc}',
+             {'units': 'seconds'}),
+        meta('slant_range_LUT', rg_lut.data, f'slant_range {desc}',
+             {'units': 'seconds'}),
+    ]
+    doppler_correction_group = dst_h5.require_group('/CSLC/grids/corrections/doppler')
```

Comment:
This contribution is the sum of the geometrical and steering Doppler. We should specify it in the variable name. Rename it to something like `geometry_steering_doppler`?

Reply:
Do the datasets have any descriptions? If not, we should add one for completeness

Reply:
Adopted your suggested name.

They do have descriptions.